staging/src/k8s.io/client-go/tools/cache/shared_informer.go

```diff
@@ -214,8 +214,8 @@ func NewSharedIndexInformer(lw ListerWatcher, exampleObject runtime.Object, defa
 		indexer:                         NewIndexer(DeletionHandlingMetaNamespaceKeyFunc, indexers),
 		listerWatcher:                   lw,
 		objectType:                      exampleObject,
-		resyncCheckPeriod:               defaultEventHandlerResyncPeriod,
-		defaultEventHandlerResyncPeriod: defaultEventHandlerResyncPeriod,
+		resyncCheckPeriod:               checkResyncPeriod(defaultEventHandlerResyncPeriod),
```

Comment:
A far fetch use case of having a resyncCheckPeriod < 1s:

One event handler has resyncPeriod=1.5s, another one has resyncPeriod=2s, then if we set resyncCheckPeriod=0.5s, then both handlers will resync timely. If the minimum resync check period is 1s, then at least one of the handler won't resync with a precise period.

Does this PR fix a real production issue?

Maybe instead of validating the period > 1s, we just check if it's >0.

Reply:
Hi @changshuchao, what do think about this comment?